### PR TITLE
Made char.IsAscii public

### LINKED ISSUE #40936: API Proposal: Make char.IsAscii public
System.Char currently defines a [private `IsAscii` method](https://github.com/dotnet/runtime/blob/d8ca3562358b036afc14c6f1abc517154c98b095/src/libraries/System.Private.CoreLib/src/System/Char.cs#L82). It think it might make sense to make this public?

```diff
public readonly struct Char
{
+  public static bool IsAscii(char ch);
}
```

### PR DESCRIPTION
**Pull request by gfoidl**
Fixes https://github.com/dotnet/runtime/issues/40936

Also normalized parameters of other method to `c` (instead of `ch`, only two others).

/cc: @GrabYourPitchforks

### CODE REVIEW

File: src/libraries/System.Runtime/tests/System/CharTests.cs [hunk 1]
```diff
@@ -199,6 +199,15 @@ public void GetTypeCode_Invoke_ReturnsBoolean()
             Assert.Equal(TypeCode.Char, 'a'.GetTypeCode());
         }
 
+        [Fact]
+        public static void IsAscii_Char()
+        {
+            Assert.True(char.IsAscii(char.MinValue));
+            Assert.True(char.IsAscii('\x007f'));
+            Assert.False(char.IsAscii('\x0080'));
+            Assert.False(char.IsAscii(char.MaxValue));
```
Comment (gfoidl) [inline src/libraries/System.Runtime/tests/System/CharTests.cs:208]: Is this enough for testing, or should be use the "whole range" ala https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.Runtime/tests/System/Text/RuneTests.TestData.cs#L233
?

Comment (danmosemsft) [reply src/libraries/System.Runtime/tests/System/CharTests.cs:208]: The whole range. There is probably a similar test for Rune.IsAscii

**Comment by danmosemsft** [comment]
This needs to be also exposed in the /ref/ file where Char surface is defined.
https://github.com/dotnet/runtime/blob/2a4284b9f3eb2ad95fdf324f0249d372abff96df/docs/coding-guidelines/updating-ref-source.md

**Comment by gfoidl** [comment]
Ahhh, ref-assembly -- thanks for the hint / info.

File: src/libraries/System.Runtime/tests/System/CharTests.cs (hunk 1 above)
Comment (gfoidl) [reply src/libraries/System.Runtime/tests/System/CharTests.cs:208]: Just scanned again through the repo, looking for IsAscii-tests. Haven't found one.

I've seen that all IsAscii implementations, like https://github.com/dotnet/runtime/blob/aa5fdab9654d74bc6274c0b5d820272c8e859621/src/libraries/System.Private.CoreLib/src/System/Text/Rune.cs#L143 delegate to https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.Private.CoreLib/src/System/Text/UnicodeUtility.cs#L120

It's a trivial method (just one compare), but maybe it's better to delegate `char.IsAscii` also to this method and add tests over there (as I didn't find some for `IsAsciiCodePoint`)?

Comment (GrabYourPitchforks) [reply src/libraries/System.Runtime/tests/System/CharTests.cs:208]: Search the file https://github.com/dotnet/runtime/blob/master/src/libraries/System.Runtime/tests/System/Text/RuneTests.cs for the substring `"IsAscii"`. You'll see how we perform tests against the `Rune` APIs.

Though to be honest those tests are basically just testing specific values like `'\u0000'`, `'\u007f'`, etc., as you had here. :)

I have no problem with you leaving the test as you have it here.

**Review by GrabYourPitchforks — COMMENTED** [review]
Remember to update the ref asms! Send a ping once that's ready. Otherwise LGTM.

**Comment by gfoidl** [comment]
@GrabYourPitchforks I pushed the commit with the ref-assembly update.

> I have no problem with you leaving the test as you have it here.

I left the test as they are.

**Review by GrabYourPitchforks — APPROVED** [review]

**Pull request merged**